### ISSUE TITLE
FormSpy: checkbox never gets "active"

### ISSUE THREAD

**Issue by msageryd** [final-form/react-final-form#827]
I'm using FormSpy for some field level condition checks. It works fine for everything but checkboxes. I'm subscribing to "active" in order to get hold of the field name for the currently active field. But checkboxes does not become "active". They get new values, which updates the modified object, but active is not updated.

Is this a bug?
Are there any workarounds?

I could keep a separate list of all the values from my checkboxes and compare them to see if any of them has changed. But this is really hard work compared to just get the field name from props.active.

---
edit:
Sorry, I should have mentioned that I'm using React Native. Checkboxes might behave different in React on the web?

For now, my workaround is to tap into the onChange event in each Field component instead of trying to handle this collectively via FormSpy. I can't see any other solution.

**Comment by msageryd** [comment]
I don't know if the above is a bug. For my use case I found it much better to handle my conditions directly in the subscribed field, so I'm not affected by this behaviour. Closing this.

**Comment by erikras** [comment]
For field-level condition checks, you should use `Field`, not `FormSpy`. Checkbox weirdness can often be fixed by passing `type="checkbox"` to `Field`, even if you are using render props. This tells RFF to manage your `checked` boolean for you.

Hope this helps.

**Comment by msageryd** [comment]
Thanks Erik.
The conditions are "global", i.e. can affect and/or be affected by more than one field. Thats why I started out with FormSpy.

Life got much easier when I started to handle this with registerField + useReducer. The reducer lives at form level, and I only need to send the dispatch function down the tree to my field registrations. The dispatch function is used in in each fieldState-event where only the conditions relevant to each field are checked.

It took quite some time, but I think I'm starting to "get" final-form. It's really neat. Thank you for providing this fantastic library.
